File: src/LitXusCount.Domain/Entities/CompanyInfo.cs
namespace LitXusCount.Domain.Entities;

/// <summary>Always exactly one row (Id = 1) — no create/delete, edit-only.</summary>
public class CompanyInfo
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;
    public string? LogoUrl { get; set; }

    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public string? PostCode { get; set; }

    public string? Phone { get; set; }
    public string? Mobile { get; set; }
    public string? Email { get; set; }
    public string? Fax { get; set; }
    public string? Website { get; set; }

    public string? CompanyRegistrationNumber { get; set; }
    public string? VatRegistrationNumber { get; set; }

    public string? InvoiceNumberPrefix { get; set; }
    public string? QuoteNumberPrefix { get; set; }
    public string? TermsAndConditions { get; set; }

    public bool IsVatEnabled { get; set; }
    public string? VatTitle { get; set; }
    public bool IsItemDiscountPercentage { get; set; }

    public long? CurrencyId { get; set; }
    public Currency? Currency { get; set; }

    public long? VatPercentageId { get; set; }
    public VatPercentage? VatPercentage { get; set; }

    public long? EmailConfigId { get; set; }
    public EmailConfig? EmailConfig { get; set; }

    public DateTime? ModifiedAt { get; set; }
    public string? ModifiedBy { get; set; }
}

File: Web/admin-dashboard/src/api/settings/companyInfo.ts
import { apiClient } from "../client";

export interface CompanyInfo {
  id: number;
  name: string;
  logoUrl: string | null;
  address: string | null;
  city: string | null;
  country: string | null;
  postCode: string | null;
  phone: string | null;
  mobile: string | null;
  email: string | null;
  fax: string | null;
  website: string | null;
  companyRegistrationNumber: string | null;
  vatRegistrationNumber: string | null;
  invoiceNumberPrefix: string | null;
  quoteNumberPrefix: string | null;
  termsAndConditions: string | null;
  isVatEnabled: boolean;
  vatTitle: string | null;
  isItemDiscountPercentage: boolean;
  currencyId: number | null;
  vatPercentageId: number | null;
  emailConfigId: number | null;
}

export type CompanyInfoUpdate = Omit<CompanyInfo, "id">;

const base = "/api/settings/company-info";

export const companyInfoApi = {
  get: async (): Promise<CompanyInfo> => (await apiClient.get<CompanyInfo>(base)).data,
  edit: async (payload: CompanyInfoUpdate): Promise<CompanyInfo> =>
    (await apiClient.put<CompanyInfo>(base, payload)).data,
  uploadLogo: async (file: File): Promise<{ url: string }> => {
    const body = new FormData();
    body.append('file', file);
    return (await apiClient.post<{ url: string }>(`${base}/logo`, body)).data;
  },
};
